README.md

```diff
@@ -82,16 +107,52 @@ amending `options` according to your back end configuration:
 <script type="module">
     import * as webeid from 'web-eid.js';
 
+    const lang = navigator.language.substr(0, 2);
     const signButton = document.querySelector("#webeid-sign-button");
 
     signButton.addEventListener("click", async () => {
-        const options = {
-            postPrepareSigningUrl: window.location.origin + "/sign/prepare",
-            postFinalizeSigningUrl: window.location.origin + "/sign/sign"
-        };
-
         try {
-            const response = await webeid.sign(options);
+            const {
+                certificate,
+                supportedSignatureAlgorithms,
+            } = await webeid.getSigningCertificate({lang});
+            
+            const prepareSigningResponse = await fetch("/sign/prepare", {
+                method: "POST",
+                headers: {
+                    "Content-Type": "application/json",
+                    [csrfHeaderName]: csrfToken
+                },
+                body: JSON.stringify({certificate, supportedSignatureAlgorithms}),
+            });
+            if (!prepareSigningResponse.ok) {
+                throw new Error("POST /sign/prepare server error: " +
+                                prepareSigningResponse.status);
+            }
+            const {
+                hash,
+                hashFunction
+            } = await prepareSigningResponse.json();
+
+            const {
+                signatureAlgorithm,
+                signature,
+            } = await webeid.sign(certificate, hash, hashFunction, {lang});
```

Comment:
For consistency, signature should be first when compared to:
* **hash, hashFunction** when deconstructing prepareSigningResponse json
* **signature, signatureAlgorithm** in finalizeSigning request body

```suggestion
            const {
                signature,
                signatureAlgorithm,
            } = await webeid.sign(certificate, hash, hashFunction, {lang});
```

Reply:
Done, thanks!